src/Microsoft.Azure.Functions.Extensions.Mcp/McpToolExtensions.cs

```diff
@@ -34,16 +34,18 @@ internal static JsonElement GetPropertiesInputSchema(this IMcpTool tool)
                     writer.WriteString("type", "array");
                     writer.WritePropertyName("items");
                     writer.WriteStartObject();
-                    writer.WriteString("type", p.PropertyType);
-                    writer.WriteEndObject();
                 }
-                else
+
+                // Write type and enum at the appropriate level
+                writer.WriteString("type", p.PropertyType);
+                WriteEnumArrayIfPresent(writer, p.EnumValues);
```

Comment:
Let's refactor this into a separate method and remove the double `p.IsArray` check. 

I like the direction here, but something like the following would improve clarity and maintainability:

``` csharp
if (p.IsArray)
{
    writer.WriteString("type", "array");
    writer.WritePropertyName("items");
    writer.WriteStartObject();

    WriteTypeAndEnum(writer, p.PropertyType, p.EnumValues);

    writer.WriteEndObject();
}
else
{
    WriteTypeAndEnum(writer, p.PropertyType, p.EnumValues);
}
```

Where the method would have the highlighted code.